backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

```diff
@@ -103,4 +120,23 @@ private boolean isDietFriendly(Recipe recipe, List<String> diets, List<String> a
     }
     return true;
   }
+
+  private List<Recipe> getRandomisedRecipes(List<Recipe> recipes) {
```

Comment:
It's bad practice to modify the object passed in by the caller, because the caller might not be expecting it.
You can use Collections.shuffle(recipes); instead and take the first m numbers you need: